R/tm_g_gh_correlationplot.R

```diff
@@ -619,86 +619,93 @@ srv_g_correlationplot <- function(id,
       trt_group <- input$trt_group
       line_vars <- unique(c(input$hline_vars, input$vline_vars))
 
-      private_q <- anl_constraint()$qenv %>% teal.code::eval_code(
+      qenv <- anl_constraint()$qenv %>% teal.code::eval_code(
         code = bquote({
-          ANL_TRANSPOSED1 <- ANL %>% # nolint
+          ANL_x <- ANL %>%
+            dplyr::filter(.data[[.(param_var)]] == .(input$xaxis_param) & !is.na(.data[[.(input$xaxis_var)]]))
```

Comment:
```suggestion
            dplyr::filter(.data[[.(param_var)]] == .(input$xaxis_param) & !is.na(.data[[.(input$xaxis_var)]])) %>%
            dplyr::select(
              .(c("USUBJID", trt_group, "AVISITCD", param_var, "PARAM", input$xaxis_var, input$yaxis_var, "LOQFL", "LBSTRESC", line_vars))
            )
```

I think we can do the selection before the mutation since no other variables are included.

Reply:
same for ANL_y as well.

Reply:
this is a very good point!

Reply:
Or even better - I can select them even earlier at ANL (once instead of twice)